### ISSUE TITLE
Autoplay lag

### ISSUE THREAD

**Issue by pulsejet** [pulsejet/memories#197]
Videos take time from loading to actually playing without any indication to the user. Bad UX

**Comment by HyCore** [comment]
yup, and sometime it doesn't load and show an error as "transcoder failed"

**Comment by HyCore** [comment]
+ it's showing this error, and not playing: 

       Uncaught (in promise) DOMException: The play() request was interrupted because the media was removed from the document.  [/apps/memories/#v/19307/802695:1     ]

**Comment by pulsejet** [comment]
> yup, and sometime it doesn't load and show an error as "transcoder failed"

Please add the output from `/tmp/go-vod.log` of when this happens

**Comment by HyCore** [comment]
At the total opposite, and other video file, with the same .mkv extension doesn't have this issue:
![image](https://user-images.githubusercontent.com/48274082/202146536-35c42153-e649-4e87-952d-49b794152fa7.png)
```
2022/11/16 10:45:22 26bd4ch6noh0: new manager for /data/ncdata/__groupfolders/6/Film/08_La sirène du Mississippi.1969-x264-AC3.mkv
2022/11/16 10:45:29 26bd4ch6noh0-360p: stopping stream
2022/11/16 10:45:29 26bd4ch6noh0-360p: ffmpeg -loglevel warning -autorotate 0 -i /data/ncdata/__groupfolders/6/Film/08_La sirène du Mississippi.1969-x264-AC3.mkv -copyts -vf scale=-2:360 -maxrate 416666 -bufsize 833332 -c:v libx264 -profile:v high -preset faster -level:v 4.0 -crf 24 -c:a aac -b:a 64k -avoid_negative_ts disabled -f hls -hls_time 3 -force_key_frames expr:gte(t,n_forced*3) -hls_segment_type mpegts -start_number 0 -hls_segment_filename /tmp/go-vod/26bd4ch6noh0-1187505722/360p-%06d.ts -
2022/11/16 10:45:29 26bd4ch6noh0-360p: recv 360p-000000.ts
2022/11/16 10:45:29 26bd4ch6noh0-360p: recv 360p-000001.ts
2022/11/16 10:45:29 26bd4ch6noh0-360p: recv 360p-000002.ts
2022/11/16 10:45:29 26bd4ch6noh0-360p: recv 360p-000003.ts
2022/11/16 10:45:29 26bd4ch6noh0-360p: recv 360p-000004.ts
2022/11/16 10:45:29 26bd4ch6noh0-360p: recv 360p-000005.ts
2022/11/16 10:45:29 26bd4ch6noh0-360p: recv 360p-000006.ts
2022/11/16 10:45:29 26bd4ch6noh0-360p: recv 360p-000007.ts
2022/11/16 10:45:29 26bd4ch6noh0-360p: recv 360p-000008.ts
2022/11/16 10:45:29 26bd4ch6noh0-360p: goal satisfied: 8
2022/11/16 10:45:32 26bd4ch6noh0-360p: resuming transcoding
2022/11/16 10:45:32 26bd4ch6noh0-360p: recv 360p-000009.ts
2022/11/16 10:45:32 26bd4ch6noh0-360p: recv 360p-000010.ts
2022/11/16 10:45:32 26bd4ch6noh0-360p: recv 360p-000011.ts
2022/11/16 10:45:32 26bd4ch6noh0-360p: recv 360p-000012.ts
2022/11/16 10:45:32 26bd4ch6noh0-360p: recv 360p-000013.ts
2022/11/16 10:45:33 26bd4ch6noh0-360p: recv 360p-000014.ts
2022/11/16 10:45:33 26bd4ch6noh0-360p: goal satisfied: 14
2022/11/16 10:45:37 26bd4ch6noh0-360p: resuming transcoding
2022/11/16 10:45:37 26bd4ch6noh0-360p: recv 360p-000015.ts
2022/11/16 10:45:37 26bd4ch6noh0-360p: recv 360p-000016.ts
2022/11/16 10:45:37 26bd4ch6noh0-360p: recv 360p-000017.ts
2022/11/16 10:45:37 26bd4ch6noh0-360p: recv 360p-000018.ts
2022/11/16 10:45:37 26bd4ch6noh0-360p: recv 360p-000019.ts
2022/11/16 10:45:37 26bd4ch6noh0-360p: recv 360p-000020.ts
2022/11/16 10:45:37 26bd4ch6noh0-360p: goal satisfied: 20
```

**Comment by HyCore** [comment]
You can test by yourself here:
https://host.chauvat.com/apps/memories/s/46RH4zoDGDYQ7ts
You have 4 film, all in MKV, the only one that works well is the one on the middle right. (in french)
All the other one get some issues.

**Comment by HyCore** [comment]
last thing, how to disable transcoding to test out again?

**Comment by pulsejet** [comment]
So,
1. Looks like the reason the playback fails is [mkv is not supported in firefox at all](https://bugzilla.mozilla.org/show_bug.cgi?id=1422891) (wth). So you can't play these videos with ff, but chrome should work fine.
2. I'm still downloading one of the files to my dev environment. Apparently there's something wrong with the produced transcodes to get the failures. This might also be a firefox bug though :(

> last thing, how to disable transcoding to test out again?

Just run `occ memories video-setup` again and answer the prompts.

**Comment by pulsejet** [comment]
Okay I can reproduce the transcode error in both firefox and chrome. Regardless, falling back to non-transcoding can possibly work only in Chrome.

**Comment by HyCore** [comment]
1. yup, most mkv video seems to not work at all, but i wonder why this mkv video works: https://host.chauvat.com/s/yP3AnWbfsixHsEf
2. yeah both firefox & chrome have this issue. but i'm pretty sure that your transcoding thing can fix it, like plex was able to do ;D i've a great faith in your work

**Comment by HyCore** [comment]
also not having the name on the file while hovering it it's a bit tricky for movies when we have multiple time the same thumbnail (but that's a different issue)

**Comment by pulsejet** [comment]
Transcoding these videos should be fixed in the next version. Basically the HLS stream doesn't like multiple audio streams, which these videos have.

> 1. yup, most mkv video seems to not work at all, but i wonder why this mkv video works: (the one in the middle right on the video folder link i sent u)

No idea. Even that video doesn't play for me in firefox 🤷🏻 

> also not having the name on the file while hovering it it's a bit tricky for movies when we have multiple time the same thumbnail (but that's a different issue)

Well the bottom line is this isn't intended usage of Memories. I don't intend to make it a general-purpose media management suite; just focus on user-generated photos in particular. Personally I use Jellyfin for this and it works wonderfully.

**Comment by HyCore** [comment]
Thanks so much for Jellyfin, i just tried out and synced it with Nextcloud film folder, it's so fluid ! Would there be a way to integrate Jellyfin player to nextcloud to read video? or even the whole app as an extension? That would be dope !

**Comment by pulsejet** [comment]
> Thanks so much for Jellyfin, i just tried out and synced it with Nextcloud film folder, it's so fluid ! Would there be a way to integrate Jellyfin player to nextcloud to read video? or even the whole app as an extension? That would be dope !

Maybe. But that'll be a separate Nextcloud app ;)
